diff --git a/mkdocs/__main__.py b/mkdocs/__main__.py
index c8d40969..f32cdda1 100644
--- a/mkdocs/__main__.py
+++ b/mkdocs/__main__.py
@@ -254,7 +254,7 @@ def cli():
 @click.option('-a', '--dev-addr', help=dev_addr_help, metavar='<IP:PORT>')
 @click.option('-o', '--open', 'open_in_browser', help=serve_open_help, is_flag=True)
 @click.option('--no-livereload', 'livereload', flag_value=False, help=no_reload_help)
-@click.option('--livereload', 'livereload', flag_value=True, default=True, hidden=True)
+@click.option('--livereload', is_flag=True, default=True, hidden=True)
 @click.option('--dirtyreload', 'build_type', flag_value='dirty', hidden=True)
 @click.option('--dirty', 'build_type', flag_value='dirty', help=serve_dirty_help)
 @click.option('-c', '--clean', 'build_type', flag_value='clean', help=serve_clean_help)
